jena-arq/src/test/java/org/apache/jena/sparql/expr/TestFunctions.java

```diff
@@ -81,28 +89,35 @@
 //            test("afn:sprintf('%1$tm %1$te,%1$tY', "+nodeStr+")",NodeValue.makeString("10 14,2005")) ;
 //    }
     
-    private static void test_exprSprintf_05(String nodeStr, String... possible) {
-        String exprStr = "afn:sprintf('%1$tm %1$te,%1$tY', "+NodeValue.makeDateTime("2005-10-14T13:09:43Z").toString()+")" ;
+    private static void test_exprSprintf_05(String nodeStr) {
+        String exprStr = "afn:sprintf('%1$tm %1$te,%1$tY', "+NodeValue.makeDateTime(nodeStr).toString()+")" ;
         Expr expr = ExprUtils.parse(exprStr) ;
         NodeValue r = expr.eval(null, FunctionEnvBase.createTest()) ;
         assertTrue(r.isString()) ;
         String s = r.getString() ;
-        // Timezones! The locale data can be -1, 0, +1 from the Z day.
-        boolean b = false ;
-        for (String poss : possible ) {
-            if ( poss.equals(s) )
-                b = true ;
+        // Parse the date
+        String dtFormat = "yyyy-MM-dd'T'HH:mm:ssXXX";
+        SimpleDateFormat sdtFormat = new SimpleDateFormat(dtFormat);
+        Date dtDate = null;
+        try {
+            dtDate = sdtFormat.parse(nodeStr);
+        } catch (ParseException e) {
+            assertFalse("Cannot parse the input date string. Message:"+e.getMessage(),false);
         }
-        assertTrue(b) ;
+        // print the date based on the current timeZone.
+        SimpleDateFormat stdFormatOut = new SimpleDateFormat("MM dd,yyyy");
+        stdFormatOut.setTimeZone(TimeZone.getDefault());
+        String outDate = stdFormatOut.format(dtDate);
+        assertEquals(s,outDate);
     }
     
     // Temporary fix for JENA-1175
     // Timezone -11:00 to any timezone can be a day ahead
-    @Test public void exprSprintf_05a() { test_exprSprintf_05("2005-10-14T14:09:43-11:00",  "10 14,2005", "10 15,2005") ; }
+    @Test public void exprSprintf_05a() { test_exprSprintf_05("2005-10-14T14:09:43-11:00") ; }
     // Timezone Z to any timezone can be a day behind or a day ahead
```

Comment:
Looking at the changes, I think this PR may make that test pass. But the comment seems to indicate that we must test with Z suffix as well. Or update the comment if that makes sense.